### PR TITLE
Adds Pandas JSON reader for off-the-shelf data loading support

### PR DESCRIPTION
**Pull request by bryangalindo**
This is an extension of the Pandas `read_json` method found ([here](https://pandas.pydata.org/docs/reference/api/pandas.read_json.html#pandas.read_json)).

## Changes
- `hamilton/plugins/pandas_extensions.py`
- `tests/plugins/test_pandas_extensions.py`

## How I tested this
Successfully ran unit tests locally using the following command in the root directory: `pytest tests/`

## Notes
N/A

## Checklist
I'm unsure where to find the project documentation, so I left it unchecked.
- [x] PR has an informative and human-readable title (this will be pulled into the release notes)
- [x] Changes are limited to a single goal (no scope creep)
- [x] Code passed the pre-commit check & code is left cleaner/nicer than when first encountered.
- [x] Any _change_ in functionality is tested
- [x] New functions are documented (with a description, list of inputs, and expected output)
- [x] Placeholder code is flagged / future TODOs are captured in comments
- [x] Project documentation has been updated if adding/changing functionality.

### CODE REVIEW

File: hamilton/plugins/pandas_extensions.py [hunk 1]
```diff
@@ -229,6 +229,28 @@ def name(cls) -> str:
...
+        return [DATAFRAME_TYPE]
+
+    def load_data(self, type_: Type) -> Tuple[DATAFRAME_TYPE, Dict[str, Any]]:
+        df = pd.read_json(self.path)
+        metadata = utils.get_file_metadata(self.path)
+        return df, metadata
+
+    @classmethod
+    def name(cls) -> str:
+        return "pandas_json"
```
Comment (skrawcz) [inline hamilton/plugins/pandas_extensions.py:251]: no need to make this specific. The combination of `applicable_types()` and name is sufficient :)

File: hamilton/plugins/pandas_extensions.py [hunk 2]
```diff
@@ -229,6 +229,28 @@ def name(cls) -> str:
...
 
 
+@dataclasses.dataclass
+class PandasJsonLoader(DataLoader):
+    """Data loader for JSON files using Pandas. Note that this currently does not support the wide array of
+    data loading params (e.g., precise_float, encoding). We will be adding this in over time, but for now
+    you can subclass this or open up an issue if this doesn't have what you want."""
+
+    path: str
+
```
Comment (skrawcz) [inline hamilton/plugins/pandas_extensions.py:239]: we'll want this part to reflect all the different parameters you can pass -- https://pandas.pydata.org/docs/reference/api/pandas.read_json.html#pandas-read-json

So have a look at https://github.com/DAGWorks-Inc/hamilton/blob/main/hamilton/plugins/pandas_extensions.py#L153-L229 for example.

File: hamilton/plugins/pandas_extensions.py [hunk 3]
```diff
@@ -231,24 +242,86 @@ def name(cls) -> str:
...
-    """Data loader for JSON files using Pandas. Note that this currently does not support the wide array of
-    data loading params (e.g., precise_float, encoding). We will be adding this in over time, but for now
-    you can subclass this or open up an issue if this doesn't have what you want."""
+    """Data loader for JSON files using Pandas.
 
-    path: str
+    Disclaimer: We're exposing all the *current* params from the Pandas read_json method [1].
+    There's a chance some of these params may get deprecated or new params may be introduced.
+    In the event that the params/kwargs below become outdated, please raise an issue or submit
+    a pull request.
```
Comment (skrawcz) [inline hamilton/plugins/pandas_extensions.py:250]: ❤️

**Comment by bryangalindo** [comment]
Merged both branches into #346.

**Pull request closed without merging**